stl/inc/xstring

```diff
@@ -2792,12 +2792,13 @@ public:
         _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data); // throws
 
         if (_New_capacity < _New_size) {
-            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size());
-            const pointer _Fancyptr = _Getal().allocate(_New_capacity + 1); // throws
+            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size()) + 1;

```

Comment:
At line 2788, _New_capacity is initialized as "_BUF_SIZE - 1", which share the same meaning as `_Myres`. Now it switch to a different meaning- allocated memory, that is , `_Myres+1`.
This halfway change of meaning won't hurt correctness. However, it will lead to confusion in the future.
I guess there are other places that have the same issue. Please consider checking every update of `_New_capacity` for potential meaning changes.

Reply:
I've changed the code to increment, allocate, decrement, which preserves the original meaning and makes the diff easier to review.